mlfinlab/online_portfolio_selection/momentum/exponential_gradient.py

```diff
@@ -0,0 +1,53 @@
+# pylint: disable=missing-module-docstring
+import numpy as np
+from mlfinlab.online_portfolio_selection.online_portfolio_selection import OLPS
+
+
+class ExponentialGradient(OLPS):
+    """
+    This class implements the Exponential Gradient Portfolio strategy. It is reproduced with
+    modification from the following paper:
+    Li, B., Hoi, S. C.H., 2012. OnLine Portfolio Selection: A Survey. ACM Comput.
+    Surv. V, N, Article A (December YEAR), 33 pages. <https://arxiv.org/abs/1212.2129>.
+
+    Exponential gradient strategy tracks the best performing stock in the last period while
+    keeping previous portfolio information by using a regularization term.
+    """
+
+    def __init__(self, update_rule, eta=0.05):
+        """
+        Initializes with the designated update rule and eta, the learning rate.
+
+        :param update_rule: (str) 'MU': Multiplicative Update, 'GP': Gradient Projection,
+                                  'EM': Expectation Maximization.
+        :param eta: (float) Learning rate.
+        """
+        super().__init__()
+        self.eta = eta
+        self.update_rule = update_rule
+
+    def _update_weight(self, time):
```

Comment:
You built a very nice API. Its great that you only have to make these small changes to create a new algo.